setuptools/command/bdist_wheel.py

```diff
@@ -447,7 +456,15 @@ def run(self):
             f"{safer_version(self.distribution.get_version())}.dist-info"
         )
         distinfo_dir = os.path.join(self.bdist_dir, distinfo_dirname)
-        self.egg2dist(self.egginfo_dir, distinfo_dir)
+        if self.dist_info_dir:
+            # Use the given dist-info directly.
```

Comment:
May I suggest to add this log message?

```suggestion
            # Use the given dist-info directly.
            log.debug(f"reusing {self.dist_info_dir}")
```

The reason behind this request is that I was testing this PR locally with another project and I was in doubt if it was working or not, until I added this line...

With a log-level of `debug`, it should not show up unless the users pass `verbose=true` via `config_settings`.

(I think that `build` may skip the `prepare_metadata_for_build_wheel` hook, but `pip` uses it).